### PR TITLE
docs: refactor troubleshooting docs

### PR DESCRIPTION
**Pull request by trentm**
Refactor the troubleshooting doc (https://www.elastic.co/guide/en/apm/agent/nodejs/current/troubleshooting.html) somewhat inspired by the ordering/organization of the Java agent's troubleshooting doc (https://www.elastic.co/guide/en/apm/agent/java/current/trouble-shooting.html).

### CODE REVIEW

**Comment by trentm** [comment]
The page's table of contents is a little bit nicer with the "problems" section more obvious under a "Common problems" header:
Screen Shot 2022-01-13 at 2 07 13 PM

**Comment by trentm** [comment]
A lot of the churn in this change is from moving the "Debug mode" section up from the bottom to near the top (similar to what is in the Java troubleshooting docs).

File: docs/troubleshooting.asciidoc [hunk 1]
```diff
@@ -20,122 +20,105 @@ so that we can analyze the problem.
...
-[[no-data-sent]]
-=== No data is sent to the APM Server
+[[use-latest-agent]]
+=== Updating to latest agent version
 
-If neither errors nor performance metrics are being sent to the APM Server,
-it's a good idea to first check your log file and look for output just as the app starts.
+The Elastic Node.js APM Agent is updated frequently and releases are not
+strongly tied to other components in the Elastic Stack.  Therefore, trying to
+update to the most recently released agent version is often the recommended
```
Comment (bmorelli25) [inline docs/troubleshooting.asciidoc:30]:
```suggestion
strongly tied to other components in the Elastic Stack.  Therefore,
updating to the most recently released agent version is often the recommended
```

**Review by bmorelli25 — DISMISSED** [review]
This looks great! We recently updated the **On this Page** (OTP) widget on the right-hand side of the page to show level three headings—I like the idea of removing the manual jump list in favor of the automated OTP.

One small nit about "trying to update". Is there a reason a user wouldn't be able to just update? Either way, looks good.

File: docs/troubleshooting.asciidoc (hunk 1 above)
Comment (trentm) [reply docs/troubleshooting.asciidoc:30]: > Is there a reason a user wouldn't be able to just update?

Probably not, no. :)  I had actually copied this language from https://www.elastic.co/guide/en/apm/agent/java/current/trouble-shooting.html#trouble-shooting-use-latest-agent

Eventually, when we have a new major version of the Node.js APM agent that drops support for some older versions of node, then the user might not be able to use the latest APM agent version. But that's a stretch. I agree the "trying to" doesn't need to be in there. Thanks.

Comment (trentm) [reply docs/troubleshooting.asciidoc]: Damn. Missed opportunity to commit your suggestion with the commit message: "do or do not, there is no try"

**Comment by trentm** [comment]
> We recently updated the On this Page (OTP) widget on the right-hand side of the page to show level three headings

Ah. That perhaps explains why in a local build I didn't see the 3rd-level, but on www.elastic.co/guide I *did*.

**Review by bmorelli25 — APPROVED** [review]
LGTM. That's curious that you can't see the OTP L3 headings locally. I wonder if it's just a case of needing to update your local clone of elastic/docs? I'll need to test that locally as I'm the one that added that feature 😨

**Comment by trentm** [comment]
> That's curious that you can't see the OTP L3 headings locally.

@bmorelli25  False alarm. I just rebuilt and ran the docs locally and it looks good:

Screen Shot 2022-01-19 at 9 21 10 AM

I *had* updated my local clone of elastic/docs, but I think I had not done a full/clean rebuild of the local docs. Not sure.

**Pull request merged**